### PR TITLE
UX: Multichain: Provide Page, Header, Content, and Footer components

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

This PR introduces `Page`, `Header`, `Content`, and `Footer` components which will be used in the new Send flow, but could be used anywhere through the app where we want a fixed header and/or footer, with scrollable content in between.

## **Manual testing steps**

Nothing to manually test here other than verifying that the `ui/components/multichain/pages/page/page.stories.js` story looks and functions properly.

## **Screenshots/Recordings**

SCR-20231007-lzuv

## **Related issues**

_Fixes #???_

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained:
  - [ ] What problem this PR is solving.
  - [ ] How this problem was solved.
  - [ ] How reviewers can test my changes.
- [ ] I’ve indicated what issue this PR is linked to: Fixes #???
- [ ] I’ve included tests if applicable.
- [ ] I’ve documented any added code.
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)).
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/multichain/pages/page/components/header/header.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  ...props
+}) => {
+  return (
+    <HeaderBase
+      padding={4}
+      width={BlockSize.Full}
+      className={classnames('multichain-page-header', className)}
+      startAccessory={startAccessory}
+      endAccessory={endAccessory}
+      {...props}
```
Comment (vthomas13) [inline ui/components/multichain/pages/page/components/header/header.js:25]: I think we'll need the wrapperProps from the HeaderBase, unless there's another way to enforce styling. While testing this with connections, I ran into the overflow issue again with long text in the children prop. It was fixed in #21217, if that helps at all.

Comment (darkwing) [reply ui/components/multichain/pages/page/components/header/header.js:26]: Overflow fixed here:  https://github.com/MetaMask/metamask-extension/pull/21245/commits/2611d13ec7db255b1b3cf6568281203c052ee7c5

File: ui/components/multichain/pages/page/page.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { Box } from '../../../component-library';
+import {
+  BlockSize,
+  Display,
+  FlexDirection,
+} from '../../../../helpers/constants/design-system';
+
+export const Page = ({
```
Comment (DDDDDanica) [inline ui/components/multichain/pages/page/page.js:11]: I was thinking about the location of`Page` here. I feel like we can use it in various flows and should be put outside `multichain` folder. wdyt?

Comment (darkwing) [reply ui/components/multichain/pages/page/page.js:11]: Eventually in concert with the DesignSystem team, but it's only going to be used in Multichain components for now, so it makes sense to have here.

**Review by DDDDDanica — APPROVED** [review]

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**